src/main/java/com/google/alpollo/AnalysisInfoServlet.java

```diff
@@ -0,0 +1,50 @@
+package com.google.alpollo;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.alpollo.model.AnalysisInfo;
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+
+/** Servlet retrieves or saves SongInfo objects to the database. */
+@WebServlet("/analysis-info")
+public class AnalysisInfoServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+
+  /**
+   * Making a GET request to our servlet with the desired song ID as a parameter
+   * will return that song from the database.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long songId = gson.fromJson(request.getReader(), Long.class);
+    AnalysisInfo analysisInfo = SongDataBase.getAnanlysisInfo(songId);
+
+    response.setContentType("application/json;");
+    try {
+      response.getWriter().println(gson.toJson(analysisInfo));
+    } catch (IOException e) {
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
+    }
+  }
+
+  /**
+   * Making a POST request to the server with the analysisInfo object as a
+   * parameter (object containing all the info related to our song) will send the
+   * song to the storage layer.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    AnalysisInfo analysisInfo;
+    try {
+      analysisInfo = gson.fromJson(request.getReader(), AnalysisInfo.class);
+      SongDataBase.saveAnalysisInfo(analysisInfo);
+    } catch (JsonSyntaxException | JsonIOException | IOException e) {
```

Comment:
have a look at the Amaury comment in #49, please

Reply:
After removing the `throws IOException` from both methods, Cloud Shell refuses to compile and test.

Reply:
The problem seems to be in `doGet`. Either `gson.fromJson` or `SongDataBase.getAnanlysisInfo` might throw `IOException` and are not inside the `try/catch` block. Can you try moving everything inside the try/catch and see if it compiles?